services/storage/storage.go

```diff
@@ -34,7 +34,8 @@ type VirtualHardDiskProperties struct {
 	Statuses map[string]*string `json:"statuses"`
 	//HyperVGeneration - Gets the HyperVGenerationType of the VirtualMachine created from the image. Possible values are common.HyperVGeneration_HyperVGenerationV1 and common.HyperVGeneration_HyperVGenerationV2
 	HyperVGeneration common.HyperVGeneration `json:"hyperVGeneration,omitempty"`
-	//DiskFileFormat - File format of the disk. possible values are common.DiskFileFormat_DiskFileFormatVHD and common.DiskFileFormat_DiskFileFormatVHDX
+	//DiskFileFormat - File format of the disk. possible values are common.DiskFileFormat_DiskFileFormatVHD, common.DiskFileFormat_DiskFileFormatVHDX and common.DiskFileFormat_DiskFileFormatUNKNOWN.
+	//If not specified, default value is common.DiskFileFormat_DiskFileFormatVHDX
```

Comment:
i dont really understand when will the UNKONWN play in role if the default is VHDX. Also I dont think there's anywhere in your code that you check for the value and substitute to default (VHDX).

Reply:
Discussed on a seperate chat. UNKNOWN Format is needed when a user passes in a download url and doesn't specify the file format for it. In that case we set it to unknown until we determine the format after downloading the file.